openmmtools/multistate/multistatereporter.py

```diff
@@ -1638,7 +1666,7 @@ def _write_sampler_states_to_given_file(self, sampler_states: list, iteration: i
 
         storage = self._storage_dict[storage_file]
         # Check if the schema must be initialized, do this regardless of the checkpoint_interval for consistency
-        is_periodic = True if (sampler_states[0].box_vectors is not None) else False
+        is_periodic = sampler_states[0].box_vectors is not None
```

Comment:
@ijpulidos is there a reason we aren't using the `MultiStateReporter.is_periodic` property here?

Reply:
I cannot say for sure. I guess this means that one can have a `box_vectors` variable in the reporter but not necessarily have `box_vectors` attribute in the sampler states, and vice versa. I don't know if we ever want those NOT to match. Is there a use case for such scenario?

Reply:
I suppose I'd be inclined to keep it the same in this PR, it just struck me as strange. Both lines were introduced in the same commit so I'm going to go on the assumption it was deliberate rather than an oversight.